pandas/core/internals/api.py

```diff
@@ -48,18 +60,23 @@ def make_block(
 
     values, dtype = extract_pandas_array(values, dtype, ndim)
 
+    needs_reshape = False
     if klass is None:
         dtype = dtype or values.dtype
         klass = get_block_type(values, dtype)
 
     elif klass is DatetimeTZBlock and not is_datetime64tz_dtype(values.dtype):
         # pyarrow calls get here
         values = DatetimeArray._simple_new(values, dtype=dtype)
+        needs_reshape = True
 
     if not isinstance(placement, BlockPlacement):
         placement = BlockPlacement(placement)
 
     ndim = maybe_infer_ndim(values, placement, ndim)
+    if needs_reshape:
```

Comment:
can't you just do this above?

Reply:
no.  the reshape has to be after the maybe_infer_ndim, and that has to be after the DatetimeArray._simple_new